Add linting & static checks

KellerFuchs opened the pull request:
- [x] Add Tox configuration calling into `flake8`, `pylint`, and `pydocstrings`.
- [x] Fix syntax warts found by `flake8`.
- [x] Fix issues uncovered by `pylint`.
- [x] Add docstrings.
- [x] Add mypy-powered static type checks.

KellerFuchs committed:
Make antissh installable as a module

This is expected by all the Python tooling, and makes setting up
automated tests much easier.

KellerFuchs committed:
Use Tox for linting and static analysis

KellerFuchs committed:
Correct issues/warts found by linting and static checks

KellerFuchs committed:
Minor simplification in antissh.log_chan

KellerFuchs committed:
Make check executable

KellerFuchs committed:
Add docstrings

KellerFuchs commented:
@kaniini I'm doing that as a prelude to trying to use `python-ssh2` (as I find it much easier to make that sort of changes if I have tooling that picks up simple-ish mistakes)

equinox commented on antissh.py:
```diff
@@ -27,7 +30,11 @@
+KLINE_CMD_TEMPLATE = config.get(
+    'host', 'kline_cmd',
+    fallback='KLINE 86400 *@{ip} :Vulnerable SSH daemon found on this host.  '
+             'Please fix your SSH daemon and try again later.\r\n',
```
Is it intended for this to be in two parts now?

KellerFuchs replied:
2 string literals one after the other just get concatenated, so it's the same string, just not as a hugely-long line.

kaniini commented:
I don't understand why we would use python-ssh2?  this entire application is built on top of asyncio and that would be even worse performance?

KellerFuchs commented:
@kaniini Oh, I misunderstood the lack of answer on IRC as signifying you had no opposition to trying it. python-ssh2 is still usable asynchronously, and it's 5-10× faster than asyncssh (mostly because it's a wrapper around libssh2, which is written in C).
I suggested that as a way to address the performance issue mentionned in the README.

If you think it's a bad idea, I just won't do it  :3

KellerFuchs commented:
PS: Regardless of whether python-ssh2 turns out to be a better ssh library for antissh, good linting and tests might be useful (and uncovered some warts), so I'd really want your opinion/feedback regardless.

kaniini commented:
I have no objection to this PR then.

KellerFuchs committed:
tox: Refactor dependencies

KellerFuchs committed:
tox: Add mypy-powered typechecking

KellerFuchs committed:
antissh: Add type hints for mypy

KellerFuchs committed:
Use tuples instead of lists for credentials

Lists can be mutated more easily, so are dangerous to use in default arguments.
This was caught by pylint.

KellerFuchs committed:
tox: Make pylint ignore redefined variable types

This is already checked by mypy, and pylint has some false positives
(likely because it doesn't understand scoping in `async` functions).

KellerFuchs committed:
antissh: Make GEOIP_COUNTRY_WHITELIST a frozenset

This guards against accidentally mutating the set, and should provide faster
lookup  :)

KellerFuchs commented:
@kaniini I think this should be good to go, if you are OK with it.
You (or your favorite CI service) can run `tox` to run the checks; there are still a couple of things flagged by `pylint`, but I think they are false positives (with the exception of TODO).
